pkg/sql/sqlitelogictest/tests/local/local_test: TestSqlLiteLogic_testindexorderby_nosort1000slt_good_0_test failed

pkg/sql/sqlitelogictest/tests/local/local_test.TestSqlLiteLogic_testindexorderby_nosort1000slt_good_0_test [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SQLiteLogicTestsBazel/6859871?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SQLiteLogicTestsBazel/6859871?buildTab=artifacts#/) on release-22.2.0 @ [776e9f889dec6fec9b4011e82c523c648b72254e](https://github.com/cockroachdb/cockroach/commits/776e9f889dec6fec9b4011e82c523c648b72254e):


```
Slow failing tests:
TestSqlLiteLogic_testindexorderby_nosort1000slt_good_0_test - 2066.35s

Slow passing tests:
TestSqlLiteLogic_testindexorderby1000slt_good_0_test - 4201.78s
TestSqlLiteLogic_testindexdelete100slt_good_2_test - 297.14s
TestSqlLiteLogic_testindexin1000slt_good_0_test - 230.81s
TestSqlLiteLogic_testindexbetween100slt_good_3_test - 141.49s
TestSqlLiteLogic_testindexcommute10slt_good_20_test - 56.23s
TestSqlLiteLogic_testindexorderby10slt_good_21_test - 55.76s
TestSqlLiteLogic_testindexcommute100slt_good_7_test - 54.20s
TestSqlLiteLogic_testindexcommute10slt_good_4_test - 49.25s
TestSqlLiteLogic_testindexorderby_nosort10slt_good_22_test - 21.47s
TestSqlLiteLogic_testindexorderby_nosort10slt_good_37_test - 20.52s
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSqlLiteLogic_testindexorderby_nosort1000slt_good_0_test.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

I found a memory budget exceeded error in the logs:
```
          logic.go:2762: 
               pq: streamer budget: memory budget exceeded: 35136 bytes requested, 57344 currently allocated, 76801 bytes in budget
      [07:41:17] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3668/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sqlitelogictest/tests/local-vec-off/local-vec-off_test_/local-vec-off_test.runfiles/com_github_cockroachdb_cockroach/external/com_github_cockroachdb_sqllogictest/test/index/orderby_nosort/1000/slt_good_1.test with config local-vec-off: 5297 tests, 1 failures
          logic.go:3806: 
              /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3668/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sqlitelogictest/tests/local-vec-off/local-vec-off_test_/local-vec-off_test.runfiles/com_github_cockroachdb_cockroach/external/com_github_cockroachdb_sqllogictest/test/index/orderby_nosort/1000/slt_good_1.test:23651: error while processing
          logic.go:3806: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3668/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sqlitelogictest/tests/local-vec-off/local-vec-off_test_/local-vec-off_test.runfiles/com_github_cockroachdb_cockroach/external/com_github_cockroachdb_sqllogictest/test/index/orderby_nosort/1000/slt_good_1.test:23651: too many errors encountered, skipping the rest of the input
          panic.go:522: -- test log scope end --
      test logs left over in: /artifacts/tmp/_tmp/601ec2e9eb9f63c0f7391fb9b965834f/logTestSqlLiteLogic_testindexorderby_nosort1000slt_good_1_test1800528041
      --- FAIL: TestSqlLiteLogic_testindexorderby_nosort1000slt_good_1_test (55.91s)
```
I believe the test could be run with this command:
```
make testlogic PKG=../sqllogictest/test TESTFLAGS="-bigtest -test.v" TESTS="TestSqlLiteLogic/local-vec-off/TestSqlLiteLogic_testindexorderby_nosort1000slt_good_1_test"
```
I tried running it once, but with the wrong config, and it timed out after an hour.

--- a/pkg/sql/logictest/logic.go
+++ b/pkg/sql/logictest/logic.go
@@ -1494,8 +1494,13 @@ func (t *logicTest) newCluster(
 	var randomWorkmem int
 	if t.rng.Float64() < 0.5 && !serverArgs.DisableWorkmemRandomization {
 		// Randomize sql.distsql.temp_storage.workmem cluster setting in
-		// [10KiB, 100KiB) range.
-		randomWorkmem = 10<<10 + t.rng.Intn(90<<10)
+		// [10KiB, 100KiB) range for normal tests and even bigger for sqlite
+		// tests.
+		if *Bigtest {
+			randomWorkmem = 100<<10 + t.rng.Intn(90<<10)
+		} else {
+			randomWorkmem = 10<<10 + t.rng.Intn(90<<10)
+		}
 	}
 
 	// Set cluster settings.

